test/integration/ynETH.t.sol

```diff
@@ -587,7 +587,51 @@ contract ynETHIntegrationTest is IntegrationBaseTest {
         bytes memory encodedError = abi.encodeWithSelector(ynETH.ZeroETH.selector);
         vm.expectRevert(encodedError);
         yneth.depositETH{value: 0}(address(this));
-    } 
+    }
+
+    function testPreviewRedeemAfterDepositAndRewards(uint256 _amount, uint256 _rewardAmount) public {
+        vm.assume(_amount > 0 && _amount <= 10_000 ether);
+        vm.assume(_rewardAmount > 0 && _rewardAmount <= 10_000 ether);
+
+        testDepositETH(_amount);
+
+        uint256 _userBalance = yneth.balanceOf(receiver);
+        assertTrue(_userBalance > 0, "testPreviewRedeemAfterDepositAndRewards: E0");
+        uint256 _previewRedeemBefore = yneth.previewRedeem(yneth.balanceOf(receiver));
+        assertEq(_previewRedeemBefore, _amount, "testPreviewRedeemAfterDepositAndRewards: E1");
+
+        vm.deal(address(executionLayerReceiver), _rewardAmount);
+        rewardsDistributor.processRewards();
+
+        assertTrue(yneth.previewRedeem(_userBalance) > _previewRedeemBefore, "testPreviewRedeemAfterDepositAndRewards: E2");
+    }
+
+    function testConvertToAssetsAfterDepositAndRewards(uint256 _amount, uint256 _rewardAmount) public {
+        vm.assume(_amount > 0 && _amount <= 10_000 ether);
+        vm.assume(_rewardAmount > 0 && _rewardAmount <= 10_000 ether);
+
+        testDepositETH(_amount);
+
+        uint256 _userBalance = yneth.balanceOf(receiver);
+        assertTrue(_userBalance > 0, "testConvertToAssetsAfterDepositAndRewards: E0");
+        uint256 _previewRedeemBefore = yneth.convertToAssets(yneth.balanceOf(receiver));
+        assertEq(_previewRedeemBefore, _amount, "testConvertToAssetsAfterDepositAndRewards: E1");
+
+        vm.deal(address(executionLayerReceiver), _rewardAmount);
+        rewardsDistributor.processRewards();
+
+        assertTrue(yneth.convertToAssets(_userBalance) > _previewRedeemBefore, "testConvertToAssetsAfterDepositAndRewards: E2");
```

Comment:
Strengthen this to actually asserEq expected shares.

If you have some rounding error that's really hard to reconcile you may use compareWithTreshol